c_glib/arrow-glib/compute.cpp

```diff
@@ -3136,6 +3142,179 @@ garrow_round_options_new(void)
 }
 
 
+typedef struct GArrowRoundToMultipleOptionsPrivate_ {
+  GArrowScalar *multiple;
+} GArrowRoundToMultipleOptionsPrivate;
+
+enum {
+  PROP_ROUND_TO_MULTIPLE_OPTIONS_MULTIPLE = 1,
+  PROP_ROUND_TO_MULTIPLE_OPTIONS_MODE,
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowRoundToMultipleOptions,
+                           garrow_round_to_multiple_options,
+                           GARROW_TYPE_FUNCTION_OPTIONS)
```

Comment:
I am curious why their is a GLib difference between the [implementation of `Arrow::RoundOptions`](https://github.com/apache/arrow/pull/12043/files#diff-c104a8d72db70ce577a9cf0666ab8cf1d541648a11c8c98751ee3064fd254a28R2910) and `Arrow::RoundToMultipleOptions`?

Reply:
You mean that why is `_WITH_PRIVATE` needed only for `GArrowRoundToMultipleOptions`, right?

Because `arrow::compute::RoundToMultipleOptions` uses not only raw C types (`int64_t` and `enum`)  but also `struct`/`class` (`arrow::Scalar`). Apache Arrow GLib wraps `struct`/`class` by its Apache Arrow GLib type such as  `GArrowScalar` for `arrow::Scalar`. Apache Arrow GLib types use reference count based memory management by `g_object_ref()` and `g_object_unref()`. We need to keep a reference for Apache Arrow GLib object (`GArrowScalar`) that wraps Apache Arrow C++ object (`arrow::Scalar`) for memory management.

Does this explanation answer your question?

Reply:
Yes, understood. Thank you for the thorough explanation!